lib/matplotlib/axis.py

```diff
@@ -1555,8 +1551,16 @@ def convert_units(self, x):
         try:
             ret = self.converter.convert(x, self.units, self)
         except Exception as e:
-            raise munits.ConversionError('Failed to convert value(s) to axis '
-                                         f'units: {x!r}') from e
+            # If x is already number like, converters other than
+            # DecimalConverter may raise error, so check here to walk around.
+            if munits.ConversionInterface.is_numlike(x):
+                return x
+            # Try convert again
```

Comment:
in what case can the conversion succeed on the second try but not the first?

Reply:
@anntzer Thanks for pointing this out. The second try is not needed. I have updated the code.